crates/node/src/interactive_cli/context.rs

```
@@ -134,25 +134,12 @@ impl ContextCommand {
                     c3 = "Root Hash"
                 );
 
-                // First, get a list of contexts using get_contexts and try_collect
-                let context_stream = ctx_client.get_contexts(None).await;
+                let contexts = ctx_client.get_contexts(None).await;
+                let mut contexts = Box::pin(contexts);
 
-                // Create a local handle to use within the async block that returns a stream
-                let ctx_client_ref = ctx_client;
-
-                // Use a simpler approach by iterating manually through the stream
-                let mut context_ids = Vec::new();
-                pin_mut!(context_stream);
-
-                // Manually iterate over the stream
-                while let Some(context_id_result) = context_stream.try_next().await? {
-                    context_ids.push(context_id_result);
-                }
-
-                // Process each context ID sequentially
-                for context_id in context_ids {
-                    if let Ok(Some(context)) = ctx_client_ref.get_context(&context_id) {
-                        if let Ok(Some(config)) = ctx_client_ref.context_config(&context_id) {
+                while let Some(context_id) = contexts.try_next().await? {
+                    if let Ok(Some(context)) = ctx_client.get_context(&context_id) {
+                        if let Ok(Some(config)) = ctx_client.context_config(&context_id) {
```

`let-else` can help reduce indentations here

```rust
let Ok(..) = __ else {
    continue;
};
```

for both

---

also, `?` is missing, we're potentially ignoring errors